### PR TITLE
add UBSAN image

### PR DESCRIPTION
**Pull request by jeroen**

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Thanks! Let's see if the previous one works, and what changes are needed to it, and then I'll merge this one as well.

Unfortunately I still cannot get to Jenkins, 8080 is not allowed at the airport. :(

**Comment by jeroen** [comment]
I realized I forgot an important detail. The rocker ubsan image actually has two installations: `R` and `RScript` run the non-ubsan system R. To use UBSAN you need to call `RD` and `RDScript`.

See [here](https://github.com/rocker-org/r-devel-san/blob/master/Dockerfile#L108-L112) for the logic.

Is there a way to configure the path to the R/RScript executable in r-hub or should I replace the symlinks in the image?

**Comment by gaborcsardi** [comment]
There is currently no way to specify the R executable. But it is easy to
add a way. Eg. If the RBINARY env var is set then we can use that. So how
about you add this env var?

**Comment by gaborcsardi** [comment]
OK, the other image is _almost_ working on the test server, but turns out that we also need xvfb to emulate X11: 
https://builder-test.r-hub.io/status/dotenv_1.0.1.tar.gz-3e6b533a3e3f74b91274521128680dfe
I'll fix that.

And again, for this PR, can you please set `RBINARY` properly? Thanks.

**Comment by jeroen** [comment]
OK I updated and rebased the PR.

**Comment by gaborcsardi** [comment]
You also need to install xvfb. Or is that already installed? 

See e.g. https://github.com/r-hub/rhub-linux-builders/commit/2929d7b222da9da6a76b617b2238046862af4e50

**Comment by gaborcsardi** [comment]
Btw. what kind of image is this? Debian or Ubuntu? I am asking because we need to create an entry for it here: https://github.com/r-hub/rhub-frontend/blob/master/public/data/platforms.json

**Comment by jeroen** [comment]
Yes [xvbf](https://github.com/rocker-org/r-devel-san/blob/master/Dockerfile#L57) is installed.

All rocker images are based off debian:testing.

**Comment by gaborcsardi** [comment]
In that case you need to do a bit more I think. E.g. https://github.com/r-hub/rhub-linux-builders/blob/2929d7b222da9da6a76b617b2238046862af4e50/debian-gcc/Dockerfile#L15-L18

**Comment by jeroen** [comment]
OK so I added that. BTW I think those things happen automatically if you use `adduser` instead of `useradd`?

**Comment by gaborcsardi** [comment]
AFAIR `adduser` is in a separate package that might not be installed:
https://packages.debian.org/stretch/adduser

Thanks!

**Comment by gaborcsardi** [comment]
I'll make a comment here when it's ready.

File: rocker-gcc-san/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM rocker:r-devel-san
```
Comment (gaborcsardi) [inline rocker-gcc-san/Dockerfile:1]: Looks like this should be `rocker/r-devel-san:latest`.

Comment (jeroen) [reply rocker-gcc-san/Dockerfile:1]: Ok can you fix it for me? I'm in the train now.

Comment (gaborcsardi) [reply rocker-gcc-san/Dockerfile:1]: Yeah, I fixed it already. Just FYI.

File: rocker-gcc-san/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+RUN useradd docker \
+	&& mkdir /home/docker \
+	&& chown docker:docker /home/docker \
+	&& addgroup docker staff
+
+ENV RHUB_PLATFORM linux-x86_64-rocker-gcc-san
```
Comment (gaborcsardi) [inline rocker-gcc-san/Dockerfile:8]: This should refer to sg from here: https://github.com/r-hub/sysreqsdb/tree/master/platforms

I fixed this, too.

**Comment by gaborcsardi** [comment]
Another note: I also need `curl` (the binary). Looks like it is in the RPM images, but not in rocker.

**Comment by gaborcsardi** [comment]
OK, this seems to work now, it uses RD: 
https://builder.r-hub.io/status/parsedate_1.1.2.tar.gz-89f081e27a62d63d0aed29129288e426

**Comment by jeroen** [comment]
I don't think it works, you should be seeing `-fsanitize` flags passed to the compiler when building packages.

**Comment by gaborcsardi** [comment]
It is only used for `R CMD check`, and not for the installation of dependencies. Is that a problem?

**Comment by jeroen** [comment]
I am not sure, you might get linking errors when mixing the two builds.

**Comment by gaborcsardi** [comment]
OK, we can use it everywhere then.... I was just worried that it would make everything _really_ slow.

**Comment by jeroen** [comment]
Well maybe it works, not sure.

Also I am not sure `CMD check` will automatically detect UBSAN messages. I'm quite sure it doesn't actually...

**Comment by gaborcsardi** [comment]
OK, let's run everything with `RD` now, and we'll see if it is very slow: https://builder.r-hub.io/status/parsedate_1.1.2.tar.gz-d0ccdf2d36c9fc51f269f803bc83f5cd

**Pull request merged**